numpy/random/setup.py

```diff
@@ -40,23 +44,145 @@ def generate_libraries(ext, build_dir):
         defs.append(("NPY_NEEDS_MINGW_TIME_WORKAROUND", None))
 
     libs = []
+    defs.append(('NPY_NO_DEPRECATED_API', 0))
     # Configure mtrand
-    config.add_extension('mtrand',
-                         sources=[join('mtrand', x) for x in
-                                  ['mtrand.c', 'randomkit.c', 'initarray.c',
+    config.add_extension('_mtrand',
+                         sources=[join('_mtrand', x) for x in
+                                  ['_mtrand.c', 'randomkit.c', 'initarray.c',
                                    'distributions.c']]+[generate_libraries],
                          libraries=libs,
-                         depends=[join('mtrand', '*.h'),
-                                  join('mtrand', '*.pyx'),
-                                  join('mtrand', '*.pxi'),],
+                         depends=[join('_mtrand', '*.h'),
+                                  join('_mtrand', '*.pyx'),
+                                  join('_mtrand', '*.pxi'),],
                          define_macros=defs,
                          )
 
-    config.add_data_files(('.', join('mtrand', 'randomkit.h')))
+    config.add_data_files(('.', join('_mtrand', 'randomkit.h')))
     config.add_data_dir('tests')
 
-    return config
+    ##############################
+    # randomgen
+    ##############################
+
+    # Make a guess as to whether SSE2 is present for now, TODO: Improve
```

Comment:
Is this the best way?  It is possible to read this from a numpy config somewhere in the build?

Reply:
This is for you @mattip or someone else with more knowledge of the build system.

Reply:
`numpy/config.h` will have `HAVE_EMMINTRIN_H`. Can we use `#include "numpy/config.h"` in `dSMFT.h`?

Reply:
We can easily modify the DSFMT source file with something like

```c
include "numpy/config.h"

#ifdef HAVE_EMMINTRIN_H
#def HAVE_SSE2 HAVE_EMMINTRIN_H
#else
#def HAVE_SSE2 0
#endif
```